lease.go

```diff
@@ -184,13 +182,13 @@ func (lm *FileLeaseManager) createOrUpdatePeer(username, pubKey string) (*WgReco
 	lm.wgRecords[username] = &WgRecord{
 		PubKey:  pubKey,
 		IP:      ipnet.IP,
-		expires: time.Now().Add(lm.leaseTime),
+		expires: expiry,
```

Comment:
I think that this could also have a zero value: https://godoc.org/golang.org/x/oauth2#Token, which will not happen in our case, but do you think it would be valuable to have a basic check on the expiry time before we assign a lease?

Reply:
It's even optional here: https://tools.ietf.org/html/rfc7662#section-2.2
What do we want to do in that case?
As it stands, it will lease an address that immediately expires?

Reply:
I might be overthinking this, I guess worst case scenario you get an address that expires and gets cleaned after `LeaserSyncInterval` time which is fine(?)

Reply:
A token that doesn't expire is not something wiresteward can work with but it can only happen if you configure the oauth2 application in this way.

I suppose we can detect it and return an http error without a lease. It's probably better to at lease return some meaningful error so that users are aware of the problematic setup?